lib/handler/pool/state.ex

```diff
@@ -54,59 +62,52 @@ defmodule Handler.Pool.State do
           {:ok, t(), reference()} | {:reject, exception()}
   def start_worker(state, fun, opts, from_pid) do
     bytes_requested = max_heap_bytes(opts)
-    name = task_name(opts)
 
     with :ok <- check_committed_resources(state, bytes_requested),
-         {:ok, ref, task_pid} <- kickoff_new_task(state, fun, opts) do
-      new_state = commit_resources(state, ref, bytes_requested, from_pid, task_pid, name)
-
+         {:ok, ref, worker} <- kickoff_new_task(state, fun, opts, from_pid) do
+      new_state = commit_resources(state, ref, worker)
       {:ok, new_state, ref}
     end
   end
 
-  def kill_worker(%State{pool: %Pool{delegate_to: pool}} = state, task_name)
-      when not is_nil(pool) do
-    if has_worker_named(state, task_name) do
-      {:ok, number_killed} = Pool.kill(pool, task_name)
-      {:ok, number_killed, state}
-    else
-      {:ok, 0, state}
-    end
+  @spec kill_worker(t(), String.t()) :: {:ok, t(), non_neg_integer()}
+  def kill_worker(state, task_name) do
+    Enum.reduce(state.workers, {:ok, state, 0}, fn
```

Comment:
Is this function meant to be part of the module's API? Or is it public incidentally for testing purposes? If it's the former, would be nice to have a function doc :+1:

Reply:
It is meant to be public (it was added back in #4). I'm planning to add docs for this in one of my follow-on PRs